etl/scripts/faostat/create_new_snapshots.py

```diff
@@ -0,0 +1,377 @@
+# Base URL of API, used to download metadata (about countries, elements, items, etc.).
+API_BASE_URL = "https://fenixservices.fao.org/faostat/api/v1/en/definitions/domain"
+# URL of walden repos and of this script (just to be included to walden index files as a reference).
+GIT_URL_TO_WALDEN = "https://github.com/owid/walden/"
+GIT_URL_TO_THIS_FILE = f"{GIT_URL_TO_WALDEN}blob/master/ingests/faostat.py"
+
+
+def create_snapshot_metadata_file(metadata: Dict[str, Any]) -> None:
+    # Path to new snapshot folder.
+    snapshot_dir_path = Path(SNAPSHOTS_DIR / metadata["namespace"] / metadata["version"])
+    # Path to new snapshot metadata file.
+    snapshot_file_path = (snapshot_dir_path / metadata["short_name"]).with_suffix(f".{metadata['file_extension']}.dvc")
+
+    # Ensure new snapshot folder exists, otherwise create it.
+    snapshot_dir_path.mkdir(exist_ok=True)
+
+    # Create metadata file for current domain dataset.
+    with open(snapshot_file_path, "w") as f:
+        ruamel.yaml.dump({"meta": metadata}, f, Dumper=ruamel.yaml.RoundTripDumper)
+
+
+class FAODataset:
+    namespace: str = NAMESPACE
+
+    def __init__(self, dataset_metadata: Dict[str, Any]):
+        """[summary]
+
+        Args:
+            dataset_metadata (dict): Dataset raw metadata.
+        """
+        self._dataset_metadata = dataset_metadata
+        self._dataset_server_metadata = self._load_dataset_server_metadata()
+
+    def _load_dataset_server_metadata(self) -> dict:
+        # Fetch only header of the dataset file on the server, which contains additional metadata, like last
+        # modification date.
+        head_request = requests.head(self.source_data_url)
+        dataset_header = head_request.headers
+        return cast(dict, dataset_header)
+
+    @property
+    def publication_year(self):
+        return self.publication_date.year
+
+    @property
+    def publication_date(self) -> dt.date:
+        return dt.datetime.fromisoformat(self._dataset_metadata["DateUpdate"]).date()
+
+    @property
+    def modification_date(self) -> dt.date:
+        last_update_date_str = self._dataset_server_metadata["Last-modified"]
+        last_update_date = parser.parse(last_update_date_str).date()
+        return last_update_date
+
+    @property
+    def short_name(self):
+        return f"{self.namespace}_{self._dataset_metadata['DatasetCode'].lower()}"
+
+    @property
+    def source_data_url(self):
+        return self._dataset_metadata["FileLocation"]
+
+    @property
+    def metadata(self):
+        """
+        Snapshot-compatible view of this dataset's metadata.
+
+        """
+        if self._dataset_metadata["DatasetDescription"] is None:
+            # Description is sometimes missing (e.g. in faostat_esb), but a description is required in index.
+            self._dataset_metadata["DatasetDescription"] = ""
+            print(f"WARNING: Description for dataset {self.short_name} is missing. Type one manually.")
+        return {
```

Comment:
Nitpick - Is there a reason why return dict and not `SnapshotMeta`? If we could use it we could leverage its `.uri` method.

Reply:
Good idea, I've just done that (to avoid making many changes, I created an additional property).